compiler/AST/symbol.cpp

```diff
@@ -1531,7 +1524,23 @@ VarSymbol *new_StringSymbol(const char *str) {
   // DefExpr(s) always goes into the module scope to make it a global
   stringLiteralModule->block->insertAtTail(stringLitDef);
 
-  CallExpr* ctorCall = new CallExpr(PRIM_MOVE, new SymExpr(s), ctor);
+  // Unresolved sym exprs will be handled specially in normalize()
+  Expr* newFirst = NULL;
+  if (dtString->symbol != NULL) {
+    newFirst = new SymExpr(dtString->symbol);
+  } else {
+    newFirst = new UnresolvedSymExpr("string");
+  }
+
+  CallExpr *initCall = new CallExpr(PRIM_NEW,
+      newFirst,
+      castTemp,
+      new_IntSymbol(strLength),   // length
+      new_IntSymbol(strLength ? strLength+1 : 0)); // size, empty string needs 0
+  initCall->insertAtTail(gFalse); // owned = false
+  initCall->insertAtTail(gFalse); // needToCopy = false
```

Comment:
I'm not sure why these aren't included in the CallExpr when you create it, since it already takes a line per argument.

Reply:
Does the CallExpr constructor only accept five arguments?